src/mqt/ddsim/qasmsimulator.py

```diff
@@ -73,22 +74,56 @@ def target(self):
     def max_circuits(self):
         return None
 
-    def run(self, quantum_circuits: QuantumCircuit | list[QuantumCircuit], **options: dict[str, Any]) -> DDSIMJob:
+    @staticmethod
+    def _bind_parameters(qc: QuantumCircuit, values: Sequence[float] | None) -> QuantumCircuit:
+        if values is None:
+            values = []
+
+        if len(qc.parameters) != len(values):
+            msg = "The number of parameters in the circuit does not match the number of parameters provided."
+            raise AssertionError(msg)
+
+        return qc.bind_parameters(dict(zip(qc.parameters, values))) if values else qc
+
+    def run(
+        self,
+        quantum_circuits: QuantumCircuit | list[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]] | None = None,
+        **options,
+    ) -> DDSIMJob:
         if isinstance(quantum_circuits, QuantumCircuit):
             quantum_circuits = [quantum_circuits]
 
         job_id = str(uuid.uuid4())
-        local_job = DDSIMJob(self, job_id, self._run_job, quantum_circuits, **options)
+        local_job = DDSIMJob(self, job_id, self._run_job, quantum_circuits, parameter_values, **options)
         local_job.submit()
         return local_job
 
     def _validate(self, quantum_circuits: list[QuantumCircuit]) -> None:
         pass
 
-    def _run_job(self, job_id: int, quantum_circuits: list[QuantumCircuit], **options: dict[str, Any]) -> Result:
+    def _run_job(
+        self,
+        job_id: int,
+        quantum_circuits: list[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]] | None,
+        **options: dict[str, Any],
+    ) -> Result:
         self._validate(quantum_circuits)
         start = time.time()
-        result_list = [self._run_experiment(q_circ, **options) for q_circ in quantum_circuits]
+
+        if not parameter_values:
+            result_list = [self._run_experiment(q_circ, values=None, **options) for q_circ in quantum_circuits]
+        else:
+            if len(quantum_circuits) != len(parameter_values):
+                msg = "The number of circuits to simulate does not match the size of the parameter list."
+                raise AssertionError(msg)
+
+            result_list = [
+                self._run_experiment(q_circ, values, **options)
+                for q_circ, values in zip(quantum_circuits, parameter_values)
+            ]
```

Comment:
Instead of binding the parameters in the `_run_experiment` method (which, in turn, requires changes in all derived simulators), you can already bind the parameters here and only pass the instantiated circuits to the `_run_experiment` method. This should simplify quite some code throughout this PR and reduce code duplication.
This should also allow you to drop the `values` parameter from the `_run_experiment` method here and in all subclasses